server/test/signup.test.js

```diff
@@ -0,0 +1,65 @@
+import request from 'supertest';
+import { stub } from 'sinon';
+import server from '../app';
+import database from '../config/database';
+import User from '../models/userModel';
+
+describe('Sign up API test', () => {
+  let requestBody;
+  beforeEach(() => {
+    requestBody = {
+      name: 'Batman',
+      email: 'bruce@wayneEnterpise.com',
+      password: 'batman',
+    };
+  });
+
+  afterAll(() => {
+    server.close();
+    database.disconnectDB();
+  });
+
+  it('should throw error if email not passed', (done) => {
+    requestBody.email = '';
+    request(server).post('/signup').send(requestBody).expect(400, done);
+  });
+
+  it('should throw error if password not passed', (done) => {
+    requestBody.password = '';
+    request(server).post('/signup').send(requestBody).expect(400, done);
+  });
+
+  it('should throw error if email not in correct format', (done) => {
+    requestBody.email = 'bruce';
+    request(server).post('/signup').send(requestBody).expect(400, done);
+  });
+
+  it('should throw error if email not in correct format', (done) => {
+    requestBody.email = 'bruce';
+    request(server).post('/signup').send(requestBody).expect(400, done);
+  });
+
+  it('should throw error if email is existing', (done) => {
+    const finOneStub = stub(User, 'findOne').returns('email');
+    const saveStub = stub(User.prototype, 'save').returns(null);
+    request(server).post('/signup').send(requestBody).expect(400)
+      .end((err) => {
+        if (err) return done(err);
+        finOneStub.restore();
+        saveStub.restore();
+        return done();
+      });
+  });
+
+  it('should save user if evrything is correct', (done) => {
+    const findOneStub = stub(User, 'findOne').returns(null);
```

Comment:
user and save instead of findonestub and savestub?

Reply:
stubbing is on a function so I guess user won't make much sense. I will  change it to findone and save.

Reply:
findoneuser?

Reply:
Yes we can go for that